### ISSUE TITLE
7.1 surround

### ISSUE THREAD

**Issue by kdpkke** [moonlight-stream/moonlight-qt#355]
Hello guys,

When I activate the "surround" audio option, would I  get true 7.1 surround sound or is it up to 5.1?

Thank you in advance

**Comment by cgutman** [comment]
I've added 7.1 channel support in the shared moonlight core library: https://github.com/moonlight-stream/moonlight-common-c/commit/85de16b41bca7ece0a6e0c69acdb76a5da0e79e0

I've also added 7.1 channel support to Moonlight PC in:
- Settings UI and SDL audio renderer (default for Linux): https://github.com/moonlight-stream/moonlight-qt/commit/cd7649a66640a1f9c75bbc03b617c44dd47fdde1
- SLAudio (default for Steam Link, not verified): https://github.com/moonlight-stream/moonlight-qt/commit/524e92c81ae43a1341222a47216020a258a04b39
- libsoundio (default for Windows and macOS, not verified): https://github.com/moonlight-stream/moonlight-qt/commit/ea62e33e51a351324eedc8e69212158a60dac37b

and for mobile platforms too:
- iOS/tvOS (not verified): https://github.com/moonlight-stream/moonlight-ios/commit/c7a7fc3fbc271fafd36e309de3e15a763e791136
- Android (not verified): https://github.com/moonlight-stream/moonlight-android/commit/49a1524f4f2b05ee00bfae69d59685a63e5cc30d and https://github.com/moonlight-stream/moonlight-android/commit/7386eb2a7850bd37c83120db98b7e621d6a941d0

I don't have a 7.1 (or even 5.1) system so I can only verify the channel mappings by ensuring it's correctly downmixed to stereo. You verified the SDL audio renderer works correctly for me already (thanks!). Once you validate libsoundio, I'll close this issue as fixed (unless you also have a Steam Link you could test too).

**Comment by dhhuston** [comment]
I am currently running AppleTV 4k and I am only decoding a stereo stream.  I verified the logs for the host side, it is sending out at least 6 channels of audio to the device.  Works fine on the nVidia Shield and another computer on my network to at least check my host computer's configuration.

**Comment by kurac338** [comment]
Hello, I believe this issue should be reopened.
Last week I have acquired a 7.1.4 surround setup (Samsung HW-Q800C soundbar paired with 9500S rear sattelites), and can confirm that 5.1 works as expected, however, 7.1 does not.

7.1 audio is broken and distorted. I don't know how best to describe it. Channels are not mapped properly, and the sound itself is not represented correctly. There are a lot of artifacts, cracks, pops, etc.

I have tried to configure the Steam Surround virtual speaker for 7.1, but it did not fix the issue. By default, the Steam driver was configured for 5.1.

As Moonlight relies on external libraries, and a combination of Sunshine and Steam virtual speakers, I don't know where I should start diagnosing the issue; so any feedback/hints/inputs would be appreciated so I may try to scope the root cause.

cc: @cgutman
